enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -35,11 +50,11 @@
 
     private final EnricherConfig config;
     private final String name;
-    private EnricherContext buildContext;
+    private MavenEnricherContext buildContext;
 
     protected Logger log;
 
-    public BaseEnricher(EnricherContext buildContext, String name) {
+    public BaseEnricher(MavenEnricherContext buildContext, String name) {
         this.buildContext = buildContext;
         // Pick the configuration which is for us
         this.config = new EnricherConfig(buildContext.getProject().getProperties(),
```

Comment:
I think this is the only spot where the project is required, so you could easily replace it with `EnricherContext.getProperties()` and so get rid of the MavenEnricherContetxt.

Reply:
The is `getProject()` below which you should check, too.

Reply:
AFAIS getProject() is used only in the ScmEnricher and IssueEnricher for extracting relevant data from the pom. I think its ok for them to cast to a MavenEnricherContext (after an instanceof check).